src/helpers.coffee

```diff
@@ -107,6 +107,10 @@ buildLocationData = (first, last) ->
     first_column: first.first_column
     last_line: last.last_line
     last_column: last.last_column
+    range: [
+      first.range[0]
+      last.range[1]
```

Comment:
Could we just add `+ 1` here to make all the ranges always exclusive?

Reply:
@GeoffreyBooth I think you're right that it makes sense to have ranges always be exclusive, as it's new for us and Babel (and seemingly all other JS ASTs, judging by a quick survey on [astexplorer.net](https://astexplorer.net)) use exclusive ranges

It seems like the single place where we specify a `range[1]` initially for our location data is in `makeLocationData()` (in `lexer.coffee`), so I updated there to use an exclusive end `range`

Then looks like Jison's `ranges` are totally unopinionated about whether ranges are inclusive/exclusive, it just passes through whatever we've set as `range`s on the tokens. And so then here we can just pass through `last.range[1]` which should already be exclusive